### PR TITLE
Update readme file

### PR DESCRIPTION
**Pull request by izmalk**
I reviewed the readme file and suggested a few improvements and updates. 
Let's review these changes so I can implement similar updates for the VM version of the charm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,101 +7,115 @@
...
 
+Use the same action without a password parameter to randomly generate a password for the operator user.
+
 ### Storage support
 
-Currently, the Charmed Kafka K8s Operator makes use of a 10GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+Currently, the Charmed Kafka K8s Operator makes use of a 10 GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
 
-This is used for logs storage, mounted on `/logs/kafka`
+This is used for logs storage, mounted on `/logs/kafka`.
```
[the remark this answers is not in the document]
Comment (izmalk) [reply README.md]: Fixed in https://github.com/canonical/kafka-k8s-operator/pull/142/commits/5be0cccc681ba413b4c1b5c57adb41b60cee5d0b 
Can you clarify what is `log-data`?

File: README.md [hunk 2]
```diff
@@ -7,101 +7,117 @@
...
 ## Relations
 
-Supported [relations](https://juju.is/docs/olm/relations):
+The Charmed Kafka K8s Operator supports Juju [relations](https://juju.is/docs/olm/relations) for interfaces listed below.
 
-#### `kafka_client` interface:
+#### The kafka_client interface
 
-The `kafka_client` interface is used with the `data-integrator` charm. This charm allows to automatically create and manage product credentials needed to authenticate with different kinds of data platform charmed products:
+The `kafka_client` interface is used with the `data-integrator` charm. This charm allows to automatically create and manage product credentials needed to authenticate with different kinds of data platform charmed products.
```
Comment (deusebio) [inline README.md:98]:
```suggestion
The `kafka_client` interface is used with the `data-integrator` charm, which upon relation automatically provides credentials and endpoints for connecting to the desired product.
```

I would add a link to the Data Integrator charm

**Review by deusebio — COMMENTED** [review]
The edits are fine. I have a few minor comments, but overall I don't have much concern with the changes.

But I'm rather wondering if we don't want to consistently re-think the readme? Right now it seems a bit of a copy and paste from the docs, and I would honestly believe that we could reference the documentation rather than copying. 

We created the README before the documentation existed, and when we created the documentation in Discourse we actually did a bit of copying from the README. Maybe it is now time to re-think what we want the README to do. 

I would honestly have this conversation now before scaling this changes to the other readme, such that we optimize the work.

File: README.md [hunk 3]
```diff
@@ -7,101 +7,117 @@
 
 ## Overview
 
-The Charmed Kafka K8s Operator delivers automated operations management from day 0 to day 2 on the [Apache Kafka](https://kafka.apache.org) event streaming platform deployed on top of a [Kubernetes cluster](https://kubernetes.io/). It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
+The Charmed Kafka K8s Operator delivers automated operations management [from Day 0 to Day 2](https://codilime.com/blog/day-0-day-1-day-2-the-software-lifecycle-in-the-cloud-age/) on the [Apache Kafka](https://kafka.apache.org) event streaming platform deployed on top of a [Kubernetes cluster](https://kubernetes.io/). It is an open source, end-to-end, production ready data platform on top of cloud native technologies.
 
-The Kafka K8s Operator can be found on [Charmhub](https://charmhub.io/kafka-k8s) and it comes with features such as:
+The Charmed Operator can be found on [Charmhub](https://charmhub.io/kafka-k8s) and it comes with features such as:
```
[the remark this answers is not in the document]
Comment (izmalk) [reply README.md:12]: Yes, that's the name. That does not prevent us from calling it a charmed operator. In this case, we are referencing THE charmed operator mentioned earlier just not to repeat ourselves too much.

File: README.md (hunk 2 above)
Comment (izmalk) [reply README.md:98]:
```suggestion
The `kafka_client` interface is used with the [Data Integrator](https://charmhub.io/data-integrator) charm, which upon relation automatically provides credentials and endpoints for connecting to the desired product.
```

**Review by deusebio — APPROVED** [review]

**Pull request merged**